lib/get-category-path.js

```diff
@@ -0,0 +1,57 @@
+'use strict';
+
+var path = require('path'),
+  utility = require('./utility');
+
+
+/**
+ * Determines the category directory structure from the pattern's data
+ *
+ * @param {Object} patternObject  individual pattern's data (from /pattern-name/pattern.yml)
+ * @param {Object} options  pattern-importer options
+ *
+ * @return {String} categoryPath  internal path url to category
+ *
+ * @requires ./lib/utility
+ * @requires ./lib/category-name-converter
+ */
+exports.getCategoryPath = function (patternObject, options) {
```

Comment:
As a side note. I'm not sure i understand what this module is supposed to do.

Some of these if statements go pretty deep, 2, 3 levels deep 4 if you could the else. :-/

What does "Determine the category directory" mean?

This makes me think that the way the category are structure could be better. If this is what needs to happen to parse it.

I wouldn't be o.k. with this code in any of our repos :-/